lib/genesis/test_utils.go

```diff
@@ -98,25 +101,51 @@ func CreateTestGenesisJSONFile(t *testing.T, asRaw bool) (filename string) {
 	return filename
 }
 
-// NewTestGenesisWithTrieAndHeader generates genesis, genesis trie and genesis header
-func NewTestGenesisWithTrieAndHeader(t *testing.T) (*Genesis, *trie.Trie, *types.Header) {
-	gen, err := NewGenesisFromJSONRaw("../../chain/gssmr/genesis.json")
-	if err != nil {
-		gen, err = NewGenesisFromJSONRaw("../../../chain/gssmr/genesis.json")
+// getAbsolutePath returns the absolute path concatenated with pathFromRoot
+func getAbsolutePath(t *testing.T, pathFromRoot string) string {
+	t.Helper()
+
+	_, fullpath, _, _ := runtime.Caller(0)
+	finderPath := path.Dir(fullpath)
+
+	const searchingFor = "go.mod"
+	for {
+		filepathToCheck := path.Join(finderPath, searchingFor)
+		_, err := os.Stat(filepathToCheck)
+
+		fileNotFound := errors.Is(err, os.ErrNotExist)
+		if fileNotFound && finderPath == "/" {
+			t.Fatal(t, "cannot find project root")
+		}
+
+		if fileNotFound {
+			finderPath = path.Dir(finderPath)
+			continue
+		}
```

Comment:
nit in my Friday morning refactoring mood 😄 

```suggestion
		if fileNotFound {
			previousFinderPath := finderPath
			finderPath = path.Dir(finderPath)
			if finderPath == previousFinderPath {
				t.Fatal(t, "cannot find project root")
			}
			continue
		}
```